src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/HandleNotDeallocatedResourcesAction.java

```diff
@@ -31,8 +31,8 @@ public class HandleNotDeallocatedResourcesAction<T extends HaFlowPathSwappingFsm
     @Override
     public void perform(S from, S to, E event, C context, T stateMachine) {
         stateMachine.getOldResources().forEach(haFlowResources ->
-                HaFlowHistoryService.using(stateMachine.getCarrier()).saveError(HaFlowHistory
-                        .withTaskId(stateMachine.getCommandContext().getCorrelationId())
+                FlowHistoryService.using(stateMachine.getCarrier()).saveError(HaFlowHistory
```

Comment:
May we not create a new instance of FlowHistoryService each iteration?

Reply:
I created a local variable for the flow history service.